packages/web-api-send-notification-runner/src/sender/notification-sender.ts

```diff
@@ -1,21 +1,96 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+import { PromiseUtils } from 'common';
 import { inject, injectable } from 'inversify';
 import { Logger } from 'logger';
+import { OnDemandPageScanRunResultProvider } from 'service-library';
+import { NotificationError, NotificationState, OnDemandPageScanResult, ScanCompletedNotification } from 'storage-documents';
 import { NotificationSenderConfig } from '../notification-sender-config';
+import { NotificationSenderWebAPIClient } from '../tasks/notification-sender-web-api-client';
+import { NotificationSenderMetadata } from '../types/notification-sender-metadata';
 
 // tslint:disable: no-null-keyword no-any
 
 @injectable()
 export class NotificationSender {
     constructor(
+        @inject(OnDemandPageScanRunResultProvider) private readonly onDemandPageScanRunResultProvider: OnDemandPageScanRunResultProvider,
+        @inject(NotificationSenderWebAPIClient) private readonly notificationSenderWebAPIClient: NotificationSenderWebAPIClient,
         @inject(NotificationSenderConfig) private readonly notificationSenderConfig: NotificationSenderConfig,
         @inject(Logger) private readonly logger: Logger,
+        @inject(PromiseUtils) private readonly promiseUtils: PromiseUtils,
     ) {}
 
     public async sendNotification(): Promise<void> {
+        let pageScanResult: OnDemandPageScanResult;
         const notificationSenderConfigData = this.notificationSenderConfig.getConfig();
-        this.logger.logInfo(`Id: ${notificationSenderConfigData.id}`);
-        this.logger.logInfo(`Reply URL: ${notificationSenderConfigData.replyUrl}`);
+
+        this.logger.logInfo(`Reading page scan run result ${notificationSenderConfigData.scanId}`);
+        pageScanResult = await this.onDemandPageScanRunResultProvider.readScanRun(notificationSenderConfigData.scanId);
+        this.logger.setCustomProperties({ scanId: notificationSenderConfigData.scanId, batchRequestId: pageScanResult.batchRequestId });
+
+        pageScanResult = await this.startSendingNotification(notificationSenderConfigData, pageScanResult);
+
+        this.logger.logInfo(`Writing page notification status to a storage.`);
+        await this.onDemandPageScanRunResultProvider.updateScanRun(pageScanResult);
+
+        this.logger.trackEvent('SendNotificationTaskCompleted');
+    }
+
+    private async startSendingNotification(
```

Comment:
rename to sendNotification, since this method resolves promise only after it sent notification, not after it triggered sending notification.